aiidalab/utils.py

```diff
@@ -165,6 +167,9 @@ def canonical_name(self):
 
     def fulfills(self, requirement):
         """Returns True if this entry fullfills the requirement."""
+        if self.version is None:
```

Comment:
Why is this here? I think this is not correct, because you need to first check that the names match.

Reply:
You are correct, I change the check to the return clause:

```
        return (
            self.canonical_name == canonicalize_name(requirement.name)
            and (self.version in requirement.specifier or self.version is None)
        )
```